tests/cli-commands/compile-tests.js

```diff
@@ -0,0 +1,129 @@
+
+describe('CompileCommand', function () {
+    const TEST_DIR = './cli-commands-test';
+    const DEFAULT_PATH = './contracts';
+    const INVALID_PATH = './unknown_folder';
+    const INVALID_FILE = 'file.txt';
+
+    let initialDir;
+    let compileCommand;
+    let pathOptionSpy;
+    let fsCreateDirSpy;
+
+    before(async () => {
+        initialDir = process.cwd();
+        fs.mkdirSync(TEST_DIR);
+        process.chdir(TEST_DIR);
+    });
+
+    beforeEach(async () => {
+        compileCommand = new CompileCommand();
+        sinon.stub(AsyncSoftExec.prototype, "exec");
+        pathOptionSpy = sinon.spy(PathOption, "process");
+        fsCreateDirSpy = sinon.spy(fileSysUtils, "createDir");
+    });
+
+    afterEach(async () => {
+        sinon.restore();
+        cleanDirectories();
+    });
+
+    after(async () => {
+        process.chdir(initialDir);
+        fs.removeSync(TEST_DIR);
+    });
+
+    function cleanDirectories () {
+        fs.removeSync('./contracts');
+        fs.removeSync('./compiled');
+    }
+
+    function createContractsFolder () {
+        fs.mkdirSync('./contracts');
+    }
+
+    function preloadContracts() {
+        fs.copyFileSync('../tests/testing-contracts/eosio.token.cpp', './contracts/eosio.token.cpp');
+        fs.copyFileSync('../tests/testing-contracts/eosio.token.hpp', './contracts/eosio.token.hpp');
+    }
+
+    it('Should initialize command properly', async () => {
+        assert(compileCommand instanceof Command);
+        assert(compileCommand.template == definition.template);
+        assert(compileCommand.description = definition.description);
+        assert(compileCommand.options == definition.options);
+        assert(compileCommand.subcommands.length == 0);
+    });
+
+    it('Should compile when valid contracts folder is specified', async () => {
+        createContractsFolder();
+        preloadContracts();
+
+        assert(await compileCommand.execute({ path: DEFAULT_PATH }));
+
+        sinon.assert.calledWith(pathOptionSpy, DEFAULT_PATH);
+        sinon.assert.calledOnceWithExactly(fsCreateDirSpy, directories.COMPILED);
+
+        let result = await pathOptionSpy.returnValues[0];
+        assert(result[0].fullPath == `${DEFAULT_PATH}/eosio.token.cpp`);
+        assert(result[0].fileName == 'eosio.token');
+    });
+
+    it('Should throw when invalid contracts folder is specified', async () => {
+        assert(await compileCommand.execute({ path: INVALID_PATH }));
+
+        sinon.assert.calledWith(pathOptionSpy, INVALID_PATH);
+        sinon.assert.notCalled(fsCreateDirSpy);
+    });
+
+    it('Should throw when specified contracts folder is empty', async () => {
+        createContractsFolder();
+
+        assert(await compileCommand.execute({ path: DEFAULT_PATH }));
+
+        sinon.assert.calledWith(pathOptionSpy, DEFAULT_PATH);
+        sinon.assert.notCalled(fsCreateDirSpy);
+    });
+
+    it('Should compile when valid contract path is specified', async () => {
```

Comment:
What is the difference with `Should compile when valid contracts folder is specified`